api/python/provisioner/validate.py

```diff
@@ -0,0 +1,180 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
+from typing import List
+
+from .inputs import (
+    ClusterParams,
+    StorageParams,
+    NodeParams
+)
+from .vendor import attr
+
+from .values import UNCHANGED
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class ValidateSetup:
+
+    @staticmethod
+    def _parse_nodes(content):
+        """
+        Parses the content of nodes.
+
+        The node names are parsed based on config-sections.
+
+        """
+        section_list = sorted(list(content))
+        output = {}
+
+        # Gives section names of only nodes : srvnode[1-12]
+        servers = [srv for srv in section_list if "srvnode" in srv
```

Comment:
@divya-vijay That logic became more more clearer now. Thanks. But there are still two problems here:
1. you iterate over sections 3 times instead of just 1. Please use just single `for` loop with multiple conditionals inside to filter enclosures, servers and clusters
2. once collected per section type you don't use that enough actually and again [iterate over section list repeating the same conditionals](--build-orig-single-iso cortx-2.0.0-791-single.iso): `'srvnode' in`, `'storage' in`, `'default' not in` etc. Instead please:
    - consider to filter default for voth servers and storages right here and assign them to a separate key inside `output` (e.g. `output[NODE_DEFAULT] = servers_defaults` and `output[STORAGE_DEFAULT] = enclosures_defaults` )
    - use these constants for other keys: `NODE` - for server, `STORAGE` - for enclosures, `CLUSTER` - for clusters (btw do we consider multiple clusters actually @ypise)
    - update logic of validation accordingly:
      - you would be able to apply defaults first without any additional filtering there
      - you would be able to iterate over servers and storages separately with any additional filtering as well